homeassistant/components/sinch/__init__.py

```diff
@@ -1 +1,92 @@
 """Component to integrate with Sinch SMS API."""
+
+from clx.xms.exceptions import UnauthorizedException, UnexpectedResponseException
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_PLATFORM, CONF_SENDER, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv, discovery
+from homeassistant.helpers.typing import ConfigType
+
+from .config_flow import check_client_connection
+from .const import (
+    ATTR_OPTIONS,
+    CONF_DEFAULT_RECIPIENTS,
+    CONF_SERVICE_PLAN_ID,
+    DEFAULT_SENDER,
+    DOMAIN,
+)
+
+PLATFORMS = [Platform.NOTIFY]
+
+CONFIG_SCHEMA = cv.removed(DOMAIN, raise_if_present=False)
+
+DEFAULT_OPTIONS = {
+    CONF_SENDER: DEFAULT_SENDER,
+    CONF_DEFAULT_RECIPIENTS: [],
+}
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Sinch SMS component."""
+
+    if Platform.NOTIFY in config:
+        for entry in config[Platform.NOTIFY]:
+            if entry[CONF_PLATFORM] == DOMAIN:
+                hass.async_create_task(
+                    hass.config_entries.flow.async_init(
+                        DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
+                    )
+                )
+    return True
+
+
+@callback
+def _async_migrate_options_from_data_if_missing(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> None:
+    data = dict(entry.data)
+    options = dict(entry.options)
+
+    if list(options) != list(DEFAULT_OPTIONS):
+        options = dict(DEFAULT_OPTIONS, **options)
+        options[CONF_DEFAULT_RECIPIENTS] = data.pop(CONF_DEFAULT_RECIPIENTS, [])
+        options[CONF_SENDER] = data.pop(CONF_SENDER, DEFAULT_SENDER)
+
+        hass.config_entries.async_update_entry(entry, data=data, options=options)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
Don't we need to create a listener which reloads the config entry when options are changed?